### PR TITLE
Fixed slow response time when clicking on general tab in the entry editor

### LINKED ISSUE #9800: improve response when clicking on the general tab
**Description**

- All linked files are on a mounted server, not on a local drive
- opening the general tab with a linked file to the distant folder results in a dealyed response
- switching between entries while general tab is open also resulsts in a delay
- opening the general tab while no file is linked shows no delay
- opening the File annotaions tab while a files is linked also results in a delay

It seems to me that the program always checks if the file exists before it returns the focus. In my [user] opinion it is not neccessary to know until I want to open the file. But to access any other field in this tab will also result in the delay.

**JabRef**
JabRef 5.9--2023-01-08--76253f1a7
Windows 11 10.0 amd64 
Java 19.0.1 
JavaFX 19+11

### PR DESCRIPTION
**Pull request by pratyushvatsalshukla**
By default Disabled Automatic Indexing Linked Files for FullText Search. It is going to improve the performance. 
UI Path : File-> Preferences -> LinkedFiles -> Automatically index all linked files for fulltext search

Screenshot 2024-01-23 194301

fixes #9800 

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/preferences/linkedfiles/LinkedFilesTabViewModel.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public void setValues() {
         mainFileDirectoryProperty.setValue(filePreferences.getMainFileDirectory().orElse(Path.of("")).toString());
         useMainFileDirectoryProperty.setValue(!filePreferences.shouldStoreFilesRelativeToBibFile());
         useBibLocationAsPrimaryProperty.setValue(filePreferences.shouldStoreFilesRelativeToBibFile());
-        fulltextIndex.setValue(filePreferences.shouldFulltextIndexLinkedFiles());
+        fulltextIndex.setValue(!filePreferences.shouldFulltextIndexLinkedFiles());
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/preferences/linkedfiles/LinkedFilesTabViewModel.java:80]: This change is wrong. It does not change the default value, it just changes the logic of how to represent the preferences in the dialog and now shows simply the opposite of the actual preferences value.
Plase note that changes to default values have to be made in JabRefPreferences.
But I believe that full text indexing is already disabled by default.

**Review by calixtus — CHANGES_REQUESTED** [review]
Hi, thanks for your interest in JabRef and open source programming.
The change you made however does not make any sense.
Please describe your reasoning for your change and why this should improve the performance.

The comments in the issue description seem to be somewhat misleading.

**Comment by calixtus** [comment]
I have to correct myself. Fulltext indexing is enabled by default. My fault. But the review is not affected by this.
Still makes me wonder, why this would impact the ui behaviour, as indexing happens in a separate cpu task and is done after a few minutes. Would be important to test the delay after indexing is done.

**Comment by pratyushvatsalshukla** [comment]
@calixtus 
Earlier, lets say i Disabled that fulltext index Checkbox manually and after my work is done, i closed the app. Now again, whenever i will open the application that Full Text Index Checkbox will be Ticked automatically.
 So each time user has to go there and untick it.
 What I did, I reversed this nature. Now after my changes, Whenever a new user will open the application, that Full Text Label Checkbox will be Disabled. And if user ticks it manually, that setting will be there until the session is closed. When user Restarts/LogIn again, then that checkbox will again be disabled.

Hope I was able to explain. Looking forward for a healthy discussion. Thanks.

**Comment by calixtus** [comment]
Hi @pratyushvatsalshukla,
@ThiloteE suggested, that the default value is set to false. But you do not set the default value by negating the presentation of the actual value in the preferences in the viewmodel when setting the values for the preferences. You set it in the `defaults` map in JabRefPreferences. Thats why we called the map `defaults`

See lines 772-773 in JabRefPreferences.java:
```java
        // Create Fulltext-Index by default
        defaults.put(FULLTEXT_INDEX_LINKED_FILES, true);
```

If you just change the behaviour of the value setter in the viewmodel this would mean the following
- user opens prefs dialog
- filePreferences.shouldFulltextIndexLinkedFiles() is true
- viewModule.setValues() sets the property to !filePreferences.shouldFulltextIndexLinkedFiles() (false)
- user hits save: viewModel.storeValues() sets filePreferences.setFulltextIndexLinkedFiles to false
- user opens prefs dialog again
- filePreferences.shouldFulltextIndexLinkedFiles() is false
- viewModule.setValues() sets the property to !filePreferences.shouldFulltextIndexLinkedFiles() (true)
- user hits save: viewModel.storeValues() sets filePreferences.setFulltextIndexLinkedFiles to true
- user opens prefs dialog again
- filePreferences.shouldFulltextIndexLinkedFiles() is true
- viewModule.setValues() sets the property to !filePreferences.shouldFulltextIndexLinkedFiles() (false)
- user hits save: viewModel.storeValues() sets filePreferences.setFulltextIndexLinkedFiles to false
...

**Comment by calixtus** [comment]
About disabling full text indexing by default: https://github.com/JabRef/jabref/issues/9800#issuecomment-1906943054

**Comment by ThiloteE** [comment]
I suggested to disable indexing as an initial form of debugging to find the real culprit for the slow response. Koppor and Processeer confirmed it could be that, but they never provided a hard proof and benchmarks, e.g. with n number of linked files, during indexing entry editor is delayed by x seconds. Later we came up with multiple proposals how we could trigger indexing differently. You chose the mose radical (but easiest) approach and disabled it completely, while automatic indexing clearly is very convenient, albeit slow in large databases. Since most users do NOT have huge databases and a preference exists that allows to disable it, I would suggest to not disable indexing by default.

**Comment by Siedlerchr** [comment]
As we discussed, this is not a solution and the issue needs more refinement. Disabling the indexer is not a solution.

**Pull request closed without merging**